enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+## Background
+
+OpenShift Monitoring is currently composed of two main systems - Platform Monitoring (PM) and User Workload Monitoring (UWM), both of which are part of OpenShift core.
+
+Platform Monitoring holds metrics about core system components running in `openshift-*` and `kube-*` namespaces. Platform Monitoring is enabled in all OpenShift installations and cannot be disabled. Some of the components scraped by PM include core-dns, etcd, kube-apiserver, image-registry, operator-lifecycle-manager etc.
+Platform Monitoring is by design an immutable stack, [which means that admins cannot extend it with additional metrics without losing the support.](https://docs.openshift.com/container-platform/4.7/monitoring/configuring-the-monitoring-stack.html#support-considerations_configuring-the-monitoring-stack)
+
+User Workload Monitoring is an optional stack managed by Cluster Monitoring Operator (CMO) and is disabled by default. Once enabled by cluster admins, it allows users to monitor namespaces that are not part of OpenShift core.
+Please refer to the User Workload Monitoring [enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/monitoring/user-workload-monitoring.md) for more details. If enabled, only one stack can be deployed. The recommended way to get “platform” data is through Thanos Querier. Alternatively, Platform Monitoring Prometheus can also federate the latest data.
+
+Recently an essential model of managing software for customers emerged at Red Hat, referred to as “Managed Services” or “Managed Tenants”. From an architectural point there are three types of such services:
+
+* Addons/Layered Services
+* SaaS
+* A combination of the first two (SaaS that requires an Addon)
+
+The way these managed services are intended to run is presented in the following diagram.
+
+![drawing](https://docs.google.com/drawings/d/e/2PACX-1vQrMBfwE8DJm3kVtC6SLVYJLEztJRABIYwvILMgtOmyyE6WqKSCqvZHksUsVUoqSZIt5E_4ro4ObGmY/pub?w=1312&h=511)
+
+We expect many more Addons to come in the following years.
+
+## Motivation
+
+Red Hat managed services are deployed in OSD clusters and run alongside user workloads.
+
+Even though there are two monitoring stacks available in OSD clusters, neither is suitable for managed services. In OSD, Platform Monitoring is exclusively owned and managed by Platform SRE. It is customized to their needs and no modifications are allowed.
+Because of this, managed tenant owners (MTOs) cannot create additional ServiceMonitors, Prometheus rules and other monitoring resources on metrics in the Platform Monitoring domain.
+
+User Workload Monitoring was designed to be primarily used by customers for monitoring their own workloads. Managed services are discouraged from using this stack since they could easily get DOS-ed by high cardinality data from end-users.
```

Comment:
Actually user workload monitoring will interfere with the middleware monitoring stack employed by add-ons today and has to be disabled in order for the managed service to be installed.

Reply:
I could be remembering that in reverse though.

Reply:
To add here, UWM doesn't just scrape customer workload metrics. Technically, it scrapes every ServiceMonitor it can find outside the platform namespaces, assuming there were no feature updates. This has been an issue in RHOAM where it exposed the Addon's monitoring to the customers which was only intended to be used by Addon/RHOAM SREs.

Reply:
Yes, this is correct and the we've already taken it into account independently of this enhancement. In 4.9 a namespace can be excluded from UWM by adding a label to it, which should solve the problem for addons.